Misc/NEWS.d/next/Library/2024-01-15-19-54-41.gh-issue-114087.Xic5vY.rst

```diff
@@ -0,0 +1 @@
+Speed up dataclasses.asdict by up to 1.35x.
```

Comment:
The performance benchmarks in the PR are on `_asdict_inner`. Could you create benchmarks on the public `.asdict` as well? I suspect performance improvements will be a a bit less and it seems fair to report only the improvements for the public interface.

```suggestion
Speed up ``dataclasses.asdict`` up to 1.35x.
```